cw_evm/lib/evm_chain_wallet.dart

```diff
@@ -0,0 +1,584 @@
+        throw EVMChainTransactionCreationException(transactionCurrency);
+      }
+    }
+
+    final pendingEVMChainTransaction = await _client.signTransaction(
+      privateKey: _evmChainPrivateKey,
+      toAddress: _credentials.outputs.first.isParsedAddress
+          ? _credentials.outputs.first.extractedAddress!
+          : _credentials.outputs.first.address,
+      amount: totalAmount.toString(),
+      gas: _estimatedGas!,
+      priority: _credentials.priority!,
+      currency: transactionCurrency,
+      exponent: exponent,
+      contractAddress:
+          transactionCurrency is Erc20Token ? transactionCurrency.contractAddress : null,
+    );
+
+    return pendingEVMChainTransaction;
+  }
+
+  Future<void> _updateTransactions() async {
+    try {
+      if (_isTransactionUpdating) {
+        return;
+      }
+
+      final isProviderEnabled = await checkIfScanProviderIsEnabled();
+
+      if (!isProviderEnabled) {
+        return;
+      }
+
+      _isTransactionUpdating = true;
+      final transactions = await fetchTransactions();
+      transactionHistory.addMany(transactions);
+      await transactionHistory.save();
+      _isTransactionUpdating = false;
+    } catch (_) {
+      _isTransactionUpdating = false;
+    }
+  }
+
+  Future<bool> checkIfScanProviderIsEnabled() async {
+    bool isEtherscanEnabled = (await sharedPrefs.future).getBool("use_etherscan") ?? true;
+    return isEtherscanEnabled;
+  }
+
+  @override
+  Future<Map<String, EVMChainTransactionInfo>> fetchTransactions() async {
+    final address = _evmChainPrivateKey.address.hex;
+    final transactions = await _client.fetchTransactions(address);
+
+    final List<Future<List<EVMChainTransactionModel>>> erc20TokensTransactions = [];
+
+    for (var token in balance.keys) {
+      if (token is Erc20Token) {
+        erc20TokensTransactions.add(_client.fetchTransactions(
+          address,
+          contractAddress: token.contractAddress,
+        ));
+      }
+    }
+
+    final tokensTransaction = await Future.wait(erc20TokensTransactions);
+    transactions.addAll(tokensTransaction.expand((element) => element));
+
+    final Map<String, EVMChainTransactionInfo> result = {};
+
+    for (var transactionModel in transactions) {
+      if (transactionModel.isError) {
+        continue;
+      }
+
+      result[transactionModel.hash] = EVMChainTransactionInfo(
+        id: transactionModel.hash,
+        height: transactionModel.blockNumber,
+        ethAmount: transactionModel.amount,
+        direction: transactionModel.from == address
+            ? TransactionDirection.outgoing
+            : TransactionDirection.incoming,
+        isPending: false,
+        date: transactionModel.date,
+        confirmations: transactionModel.confirmations,
+        ethFee: BigInt.from(transactionModel.gasUsed) * transactionModel.gasPrice,
+        exponent: transactionModel.tokenDecimal ?? 18,
+        tokenSymbol: transactionModel.tokenSymbol ?? "ETH",
```

Comment:
should not be the case for polygon

Reply:
Yes, Polygon previously had an override for it

But I've adjusted to fit the implementation